scripts/generate_schema.py

```diff
@@ -7,13 +7,23 @@
 usage: python generate_schema.py [<OUT_DIR>]
 """
 
+from typing import Any, Literal
 import json
 import sys
 from pathlib import Path
 
 from pydantic import TypeAdapter
 
 from hugr.serialization import SerialHugr
+from hugr.serialization.hugrtype import HugrType
+
+def write_schema(out_dir: Path, name_prefix: str, schema):
```

Comment:
`schema: Type[BaseModel]` ?

Reply:
in which case you need to constrain it to a type that has that classmethod. Either by defining a common baseclass or protocol for `SerialHugr, TestingHugr` etc. to share or using a `Type[SerialHugr] | Type[TestingHugr]` annotation here